tests/rptest/redpanda_cloud_tests/high_throughput_test.py

```diff
@@ -1811,10 +1823,12 @@ def _get_metrics(bench):
             OMBSampleConfigurations.E2E_LATENCY_AVG:
             [OMBSampleConfigurations.lte(target_e2e_avg)],
         }
+
         # Select number of partitions
         if partitions == "min":
             _num_partitions = self._partitions_min
-        elif partitions == "max":
+        else:
```

Comment:
Example of a runtime change to help the type checker. One of these checks will succeed but the type checker cna't know that, so `_num_partitoins` could also be `Unbond` from its PoV. Instead, always enter the last branch and use an assert to check that it has the expected value. Ends up being less code overall anyway since we can remove the explicit check above.

`Enum` could perhaps accomplish the same thing.

Reply:
I don't mind much about this random python testing code so please leave as is but I do find the original way easier to understand. One problem with that though is that the allowed values are duplicated. 

> Enum could perhaps accomplish the same thing.

I guess this would be the best as then you can avoid duplicating the different enum values, have a "default" handler and probably the typer check is also happy.

Reply:
@StephanDollberg wrote:

> I do find the original way easier to understand.

I agree, the original way is easier to understand though arguably less correct (since it admits a third state after the two checks instead of the desired 2).

The problem though is that it does not type check, so the alternative w/o changing types I think is something adding a third and final `else`:

```
        if partitions == "min":
            _num_partitions = self._partitions_min
        elif partitions == "max":
            _num_partitions = self._partitions_upper_limit
        else:
            raise RuntimeError('oh no')
```

I don't if Enum solves it exactly as it would require the type checker to know that the flow always goes down one of the two branches even though they are not exhaustive (i.e., there is no `else` clause) through value analysis.

Actuallly since we don't have switch in Python I kind of like using a dict like:

```
_num_partitions = {
   'min': self._partitions_min,
   'max': self._partitions_upper_limit
}[partitions]
```

WDYT? Not sure if it's idiomatic, but it type checks and gives a pretty good error (showing the key value, which is the important part).

Reply:
> Actuallly since we don't have switch in Python I kind of like using a dict like:

Yeah I like that.